assets/js/blocks/cart-checkout/cart/full-cart/cart-line-item-row.js

```diff
@@ -43,6 +44,20 @@ const CartLineItemRow = ( { lineItem = {} } ) => {
 		isPending: itemQuantityDisabled,
 	} = useStoreCartItem( key );
 
+	// Store item quantity in local state so the UI can update independently
+	// of store/server updates.
+	const [ localQuantity, updateLocalQuantity ] = useState( quantity );
+	// Debounce updates to the local quantity value, and update the store
+	// and server when value stablises.
+	// Note that other dependent values in the row (e.g. total) are updated
+	// independently (after debounce).
+	const [ debouncedQuantity ] = useDebounce( localQuantity, 400 );
+	useEffect( () => {
+		if ( key ) {
+			changeQuantity( debouncedQuantity );
+		}
+	}, [ key, debouncedQuantity ] );
```

Comment:
for now, this triggers an update on initial render, we don’t need that, also there is no need to check for key and watch it since it’s stable and required for this component to work, so we can update this to avoid extra render+call for each item.
```suggestion
	useEffect( () => {
		if ( quantity !== debouncedQuantity ) {
			changeQuantity( debouncedQuantity );
		}
	}, [ debouncedQuantity ] );
```

Reply:
Hmm, looks like this was a remnant/workaround from a different iteration. I've fixed this here 960357c4092421b7968328b688cbf88e8608c0b8 (of course may change if I refactor the state/debounce).